drivers/sensor/lps22df/lps22df.c

```diff
@@ -0,0 +1,356 @@
+	return 0;
+}
+
+static const struct sensor_driver_api lps22df_driver_api = {
+	.attr_set = lps22df_attr_set,
+	.sample_fetch = lps22df_sample_fetch,
+	.channel_get = lps22df_channel_get,
+#if CONFIG_LPS22DF_TRIGGER
+	.trigger_set = lps22df_trigger_set,
+#endif
+};
+
+static int lps22df_init_chip(const struct device *dev)
+{
+	const struct lps22df_config * const cfg = dev->config;
+	struct lps22df_data *data = dev->data;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps22df_id_t id;
+	lps22df_stat_t status;
+	uint8_t tries = 10;
+	int ret;
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	if (cfg->i3c.bus != NULL) {
+		/*
+		 * Need to grab the pointer to the I3C device descriptor
+		 * before we can talk to the sensor.
+		 */
+		data->i3c_dev = i3c_device_find(cfg->i3c.bus, &cfg->i3c.dev_id);
+		if (data->i3c_dev == NULL) {
+			LOG_ERR("Cannot find I3C device descriptor");
+			return -ENODEV;
+		}
+	}
+#endif
+
+	if (lps22df_id_get(ctx, &id) < 0) {
+		LOG_ERR("%s: Not able to read dev id", dev->name);
+		return -EIO;
+	}
+
+	if (id.whoami != LPS22DF_ID) {
+		LOG_ERR("%s: Invalid chip ID 0x%02x", dev->name, id.whoami);
+		return -EIO;
+	}
+
+	LOG_DBG("%s: chip id 0x%x", dev->name, id.whoami);
+
+	/* Restore default configuration */
+	if (lps22df_init_set(ctx, LPS22DF_RESET) < 0) {
+		LOG_ERR("%s: Not able to reset device", dev->name);
+		return -EIO;
+	}
+
+	do {
+		if (!--tries) {
+			LOG_DBG("sw reset timed out");
+			return -ETIMEDOUT;
+		}
+		k_usleep(50);
+
+		if (lps22df_status_get(ctx, &status) < 0) {
+			return -EIO;
+		}
+	} while (status.sw_reset);
+
+	/* Set bdu and if_inc recommended for driver usage */
+	if (lps22df_init_set(ctx, LPS22DF_DRV_RDY) < 0) {
+		LOG_ERR("%s: Not able to set device to ready state", dev->name);
+		return -EIO;
+	}
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	if (cfg->i3c.bus != NULL) {
+		/*
+		 * Enabling I3C and disabling I2C are required
+		 * for I3C IBI to work, or else the sensor will not
+		 * send any IBIs.
+		 */
+
+		ret = lps22df_i3c_interface_set(ctx, LPS22DF_I3C_ENABLE);
+		if (ret < 0) {
+			LOG_ERR("Cannot enable I3C interface");
+			return ret;
+		}
+
+		ret = lps22df_i2c_interface_set(ctx, LPS22DF_I2C_DISABLE);
+		if (ret < 0) {
+			LOG_ERR("Cannot disable I2C interface");
+			return ret;
+		}
+	}
+#else
+	ARG_UNUSED(data);
+#endif
+
+	/* set sensor default odr */
+	LOG_DBG("%s: odr: %d", dev->name, cfg->odr);
+	ret = lps22df_set_odr_raw(dev, cfg->odr);
+	if (ret < 0) {
+		LOG_ERR("%s: Failed to set odr %d", dev->name, cfg->odr);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int lps22df_init(const struct device *dev)
+{
+	if (lps22df_init_chip(dev) < 0) {
+		LOG_DBG("Failed to initialize chip");
+		return -EIO;
+	}
+
+#ifdef CONFIG_LPS22DF_TRIGGER
+	if (lps22df_init_interrupt(dev) < 0) {
+		LOG_ERR("Failed to initialize interrupt.");
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "LPS22DF driver enabled without any devices"
+#endif
```

Comment:
I don't think this can ever happen since you have

```
depends on DT_HAS_ST_LPS22DF_ENABLED
```

so you may as well drop it

Reply:
that's correct, thanks